xlators/features/locks/src/posix.c

```diff
@@ -207,24 +216,24 @@ dict_delete_domain_key(dict_t *dict, char *key, data_t *value, void *data)
     return 0;
 }
 
-void
-pl_get_xdata_requests(pl_local_t *local, dict_t *xdata)
+static void
+pl_get_xdata_requests(pl_local_t *local, dict_t *xdata, unsigned char bitfield)
 {
     if (!local || !xdata)
         return;
 
     GF_ASSERT(local->xdata == NULL);
     local->xdata = dict_copy_with_ref(xdata, NULL);
 
-    if (dict_get_sizen(xdata, GLUSTERFS_ENTRYLK_COUNT)) {
+    if (IS_BIT_SET(bitfield, 0)) {
         local->entrylk_count_req = 1;
         dict_del_sizen(xdata, GLUSTERFS_ENTRYLK_COUNT);
```

Comment:
We could improve this even more by doing a small change in `dict_del_*()` functions. We could return a boolean indicating whether the value was present or not. Using this new functionality, we can change the `dict_getn()` at line 202 by a `dict_deln()` and we won't need to do it here.